Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -239,45 +202,117 @@ void CGPathAddLineToPoint(CGMutablePathRef path, const CGAffineTransform* m, flo
 
     if (pathObj->_count + 1 >= pathObj->_max) {
         pathObj->_max += 32;
-        pathObj->_components = (pathComponent*)IwRealloc(pathObj->_components, pathObj->_max * sizeof(pathComponent));
+        pathObj->_components = (CGPathElement*)IwRealloc(pathObj->_components, pathObj->_max * sizeof(CGPathElement));
     }
 
-    pathObj->_components[pathObj->_count].type = pathComponentLineTo;
-    pathObj->_components[pathObj->_count].point.x = x;
-    pathObj->_components[pathObj->_count].point.y = y;
+    pathObj->_components[pathObj->_count].type = kCGPathElementAddLineToPoint;
+
+    pathObj->_components[pathObj->_count].points = (CGPoint*)IwCalloc(1, sizeof(CGPoint));
+    pathObj->_components[pathObj->_count].points[0] = { x, y };
 
     pathObj->_count++;
 }
 
+CGFloat _CGPathControlPointOffsetMultiplier(CGFloat angle) {
+    return (4.0f / 3.0f) * tan(angle / 4.0f);
+}
+
 /**
  @Status Caveat
  @Notes transform property not supported
 */
 void CGPathAddArcToPoint(CGMutablePathRef path, const CGAffineTransform* m, float x1, float y1, float x2, float y2, float radius) {
```

Comment:
The documentation on this is not very clear.  "If the current point and the first tangent point of the arc (the starting point) are not equal, Quartz appends a straight line segment from the current point to the first tangent point." On the other hand, the first tangent is from current point to x1, y1.  Does the reference platform bail out if current point is same as x1, y1 like the code below (it makes sense, but again,the reference doc is not clear).